pkg/crypto/certloader/server.go

```diff
@@ -134,3 +137,13 @@ func (c *WatchedServerConfig) ServerConfig(base *tls.Config) *tls.Config {
 		MinVersion: tls.VersionTLS13,
 	}
 }
+
+// appendH2Proto appends h2 to the list of protocols if it's not already there.
+func (c *WatchedServerConfig) appendH2Proto(existingProtocols []string) []string {
```

Comment:
I see no reason to make this function a method of `WatchedServerConfig`.
```suggestion
func appendH2Proto(existingProtocols []string) []string {
```

Reply:
Sure, updated.